Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/RdbmsCentralRepo.java

```diff
@@ -1496,6 +1496,59 @@ public Long getCountUniqueCaseDataSourceTuplesHavingTypeValue(CorrelationAttribu
+        if (instance != null) {
+            Long sourceObjID = instance.getFileObjectId();
+            //We know that instances have a correlation case but that correlation case may have a null ID if it is not in the CR, although it should be.
+            int correlationCaseId = instance.getCorrelationCase().getID();
+            String normalizedValue = CorrelationAttributeNormalizer.normalize(instance.getCorrelationType(), instance.getCorrelationValue());
+            Connection conn = connect();
+            PreparedStatement preparedStatement = null;
+            String tableName = CentralRepoDbUtil.correlationTypeToInstanceTableName(instance.getCorrelationType());
+            ResultSet resultSet = null;
+            try {
+                if (correlationCaseId > 0 && sourceObjID != null) {
+                    //the current case is in the CR we can ignore this case source file to ensure we don't count the current item   
+                    //this will also work regardless of the instance itself being a database instance
+                    String sql
+                            = "SELECT count(*) FROM (SELECT DISTINCT case_id FROM "
+                            + tableName
+                            + " WHERE value=? AND NOT (file_obj_id=? AND case_id=?)) AS "
+                            + tableName
+                            + "_other_case_count";
+                    preparedStatement = conn.prepareStatement(sql);
+                    preparedStatement.setString(1, normalizedValue);
+                    preparedStatement.setLong(2, sourceObjID);
+                    preparedStatement.setInt(3, correlationCaseId);
+                } else {
+                    //the current case is not in the CR so the current instance can't be so we can just count all other cases with the instance
+                    //we won't know if it exists elsewhere in this case because this case is not in the CR
+                    String sql
+                            = "SELECT count(*) FROM (SELECT DISTINCT case_id FROM "
+                            + tableName
+                            + " WHERE value=? AS "
+                            + tableName
+                            + "_other_case_count";
+                    preparedStatement = conn.prepareStatement(sql);
+                    preparedStatement.setString(1, normalizedValue);
+                }
+                resultSet = preparedStatement.executeQuery();
+                resultSet.next();
+                instanceCount = resultSet.getLong(1);
+            } catch (SQLException ex) {
+                throw new CentralRepoException("Error counting unique caseDisplayName/dataSource tuples having artifactType and artifactValue.", ex); // NON-NLS
+            } finally {
+                CentralRepoDbUtil.closeStatement(preparedStatement);
+                CentralRepoDbUtil.closeResultSet(resultSet);
+                CentralRepoDbUtil.closeConnection(conn);
```

Comment:
Alright I changed all the closing sequences in this file to close the result set first, so that when we add new ones any that are looked at for reference will be the correct order.